pkg/types/spec/errors.go

```diff
@@ -359,6 +361,26 @@ func ErrorInvalidNumberOfInfs(requestedInfs int64) error {
 	})
 }
 
+func ErrorInsufficientBatchConcurrencyLevel() error {
+	return errors.WithStack(&errors.Error{
+		Kind: ErrInsufficientBatchConcurrencyLevel,
+		Message: fmt.Sprintf(
+			"%s must be greater than or equal to %s * %s",
+			userconfig.MaxBatchSizeKey, userconfig.ProcessesPerReplicaKey, userconfig.ThreadsPerProcessKey,
+		),
+	})
+}
+
+func ErrorInsufficientBatchConcurrencyLevelInf() error {
+	return errors.WithStack(&errors.Error{
+		Kind: ErrInsufficientBatchConcurrencyLevelInf,
+		Message: fmt.Sprintf(
+			"%s must be greater than or equal to %s",
+			userconfig.MaxBatchSizeKey, userconfig.ThreadsPerProcessKey,
+		),
+	})
+}
+
```

Comment:
For instance, if the batch size is set to 8, then you need at least 8 concurrent requests to be able to fill in all those spots. And since the number of concurrent requests is determined by `processes_per_replica * threads_per_process`, that's the reason why we have that error message. If `processes_per_replica * threads_per_process` is too low, then all requests will block until the batch interval (timeout) is reached.

As for the case with Inferentia, the reason why we're only limited by `threads_per_process` is that for each process we have a separate TFS process. And so all requests directed to a single serving process will be distributed to a single TFS process (as opposed to being collected from all serving processes and directed to a single TFS) - this means each serving process must get at least as many requests as the batch size is set for the TFS process.

Reply:
That all makes sense, thanks for the detailed explanation!

Reply:
I think just the words are out of order on the error message, I can switch it tomorrow, I just wanted to run it by you to make sure